modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/ServerShowNotesManager.kt

```diff
@@ -1,165 +1,85 @@
-                val cachedNotes = getShowNotesFromResponse(cachedResponse)
-                // only download if not cached already
-                if (cachedNotes.isNullOrEmpty()) {
-                    // download to cache
-                    val request = buildNetworkShowNotesRequest(url)
-                    httpShowNotesCache.newCall(request).execute().use { response ->
-                        Timber.i("Show notes cached %s %s", episodeUuid, if (response.isSuccessful) "Successful" else "Failed")
-                    }
-                }
-            }
-        }.doOnError { throwable ->
-            LogBuffer.e(LogBuffer.TAG_BACKGROUND_TASKS, throwable, "Unable to cache show notes.")
-            Timber.e(throwable)
-        }
-    }
-
-    fun loadShowNotes(episodeUuid: String, callback: CachedServerCallback<String>?) {
-        val url = buildUrl(episodeUuid)
-
-        val uiHandler = if (callback == null) null else Handler(Looper.getMainLooper())
-
-        // load the cache version first for speed
-        loadCachedShowNotes(
-            url,
-            object : ShowNotesListener {
-                override fun complete(showNotes: String?) {
-                    if (showNotes != null && showNotes.isNotBlank()) {
-                        uiHandler?.post { callback?.cachedDataFound(showNotes) }
-                        return
-                    }
-                    // check the server for an updated version
-                    loadNetworkShowNotes(url, showNotes, uiHandler, callback)
+class ServerShowNotesManager @Inject constructor(
+    private val podcastCacheServerManager: PodcastCacheServerManager
+) {
+
+    /**
+     * Check the cache for show notes then download them if not found or update the cache.
+     */
+    fun loadShowNotesFlow(podcastUuid: String, episodeUuid: String): Flow<ShowNotesState> {
+        return flow {
+            emit(ShowNotesState.Loading)
+            var loaded = false
+            try {
+                // load the cache version first for speed
+                val showNotesCached = findShowNotesInCache(podcastUuid = podcastUuid, episodeUuid = episodeUuid)
+                if (!showNotesCached.isNullOrBlank()) {
+                    emit(ShowNotesState.Loaded(showNotesCached))
+                    loaded = true
                 }
-            }
-        )
-    }
-
-    suspend fun loadShowNotes(episodeUuid: String): String? =
-        suspendCoroutine { cont ->
-            loadShowNotes(
-                episodeUuid,
-                object : CachedServerCallback<String> {
-                    override fun cachedDataFound(data: String) {
-                        cont.resume(data)
-                    }
-
-                    override fun networkDataFound(data: String) {
-                        cont.resume(data)
-                    }
-
-                    override fun notFound() {
-                        cont.resume(null)
+                // download or update cache
```

Comment:
After it has checked the cache it tries to download the show notes from the server again. The okhttp library will respect the server five minute expiry and not call again if it's within that time. If it's been longer it will call the server. The server will either respond back with a 'not modified' or the new content. This will update the local cache and show the user the updated show notes if they have changed. Any idea how to make that into a better comment? 🤔